### PR TITLE
Implement #828 Create admin status page

### PR DESCRIPTION
**Pull request by ml729**
status_page_pic

### CODE REVIEW

File: website/src/pages/api/admin/stats.ts
Comment (notmd) [inline website/src/pages/api/admin/stats.ts:12]: Can you move this to `oasst_api_file`? Also, unify these 3 endpoints into 1. Cause these endpoints are not being used anywhere except the status page, if they are being used they may have different params

File: website/src/pages/api/admin/tasks_availability.ts
Comment (fozziethebeat) [inline website/src/pages/api/admin/tasks_availability.ts:13]: What's going on with this? are you just polling data for some dummy user? Can we instead use the current admin user?

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Thank you for putting this together.  Its a great start for a simple admin panel.

To keep the code and components tidy please also address @notmd's requests.  I agree with those requested changes.  I'll approve after those changes are made.

File: website/src/pages/api/admin/stats.ts
Comment (ml729) [reply website/src/pages/api/admin/stats.ts]: Thanks, @notmd for the feedback, I made the requested changes

File: website/src/pages/api/admin/tasks_availability.ts
Comment (ml729) [reply website/src/pages/api/admin/tasks_availability.ts]: Yes, in issue [#828](https://github.com/LAION-AI/Open-Assistant/issues/828) @andreaskoepf mentioned using a dummy user.
For using the current admin user instead, would you know how to get the current user's user ID?  session.user only contains the display name.

Comment (ml729) [reply website/src/pages/api/admin/tasks_availability.ts]: Nevermind I think I got it working using the token (I assumed the auth_method is always "local" if that's alright). Now it uses the current admin user.

File: website/src/pages/admin/status/index.tsx
Comment (notmd) [inline website/src/pages/admin/status/index.tsx:87]:
```suggestion
        <Card>
          <CardBody>
            <Text as="h1" fontSize="3xl" textAlign="center">
              /api/v1/tasks/availability
            </Text>
            <Box bg={dataBackgroundColor} borderRadius="xl" p="6" pt="4" pr="12">
              <pre id="json">
                {tasksAvailability ? JSON.stringify(tasksAvailability, null, 2) : <CircularProgress isIndeterminate />}
              </pre>
            </Box>
          </CardBody>
        </Card>
```

File: website/src/pages/admin/status/index.tsx
Comment (notmd) [inline website/src/pages/admin/status/index.tsx:96]:
```suggestion
        <Card>
          <CardBody>
            <Text as="h1" fontSize="3xl" textAlign="center">
              /api/v1/stats/
            </Text>
            <Box bg={dataBackgroundColor} borderRadius="xl" p="6" pt="4" pr="12">
              <pre id="json">{stats ? JSON.stringify(stats, null, 2) : <CircularProgress isIndeterminate />}</pre>
            </Box>
          </CardBody>
        </Card>
```

File: website/src/pages/admin/status/index.tsx
Comment (notmd) [inline website/src/pages/admin/status/index.tsx:151]:
```suggestion
      <Card>
        <CardBody>
          <Text as="h1" fontSize="3xl" textAlign="center">
            /api/v1/stats/tree_manager
          </Text>
          {treeManager ? (
            <Box>
              <Text as="h2" fontSize="2xl">
                state_counts
              </Text>
              <Box bg={dataBackgroundColor} borderRadius="xl" p="6" pt="4" pr="12">
                <pre id="json">{JSON.stringify(treeManager.state_counts, null, 2)}</pre>
              </Box>
              <TableContainer>
                <br />
                <Text as="h2" fontSize="2xl">
                  message_counts
                </Text>
                <Table variant="simple">
                  <TableCaption>Tree Manager</TableCaption>
                  <Thead>
                    <Tr>
                      <Th>Message Tree ID</Th>
                      <Th>State</Th>
                      <Th>Depth</Th>
                      <Th>Oldest</Th>
                      <Th>Youngest</Th>
                      <Th>Count</Th>
                      <Th>Goal Tree Size</Th>
                    </Tr>
                  </Thead>
                  <Tbody>
                    {treeManager.message_counts.map(
                      ({ message_tree_id, state, depth, oldest, youngest, count, goal_tree_size }) => (
                        <Tr key={message_tree_id}>
                          <Td>{message_tree_id}</Td>
                          <Td>{state}</Td>
                          <Td>{depth}</Td>
                          <Td>{oldest}</Td>
                          <Td>{youngest}</Td>
                          <Td>{count}</Td>
                          <Td>{goal_tree_size}</Td>
                        </Tr>
                      )
                    )}
                  </Tbody>
                </Table>
              </TableContainer>
            </Box>
          ) : (
            <CircularProgress isIndeterminate />
          )}
        </CardBody>
      </Card>
```

File: website/src/pages/api/admin/status.ts
Comment (notmd) [inline website/src/pages/api/admin/status.ts:17]:
```suggestion
  const [tasksAvailabilityData, statsData, treeManagerData] = await Promise.all([
    oasstApiClient.fetch_tasks_availability(currentUser),
    oasstApiClient.fetch_stats(),
    oasstApiClient.fetch_tree_manager(),
  ]);
```

**Review by notmd — CHANGES_REQUESTED** [review]
Also, could you handle the loading and error state correctly? Currently, you are showing the loading state when data is undefined which is incorrect. The data is undefined when `loading` or `error` is truthy. You may want to check the [swr docs](https://swr.vercel.app/docs/error-handling).

File: website/src/pages/admin/status/index.tsx
Comment (AbdBarho) [inline website/src/pages/admin/status/index.tsx:83]: why the `id=json`? is it referenced somewhere?

File: website/src/pages/api/admin/status.ts
Comment (AbdBarho) [inline website/src/pages/api/admin/status.ts:14]: maybe you can just call `getBackendUser`? I think it should work.

Comment (fozziethebeat) [reply website/src/pages/api/admin/status.ts:14]: +1

Please call `getBackendUser`.  That returns the most accurate user representation.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]

File: website/src/pages/api/admin/tasks_availability.ts
Comment (andreaskoepf) [reply website/src/pages/api/admin/tasks_availability.ts]: The "problem" with task-availability is that it is user and `lang` tag dependent .. e.g. a user cannot label own messages and  counts as the number of active message-trees are calculated per language (currently the same global setting applies to all languages).

Comment (ml729) [reply website/src/pages/api/admin/tasks_availability.ts]: Do you mean we should stick with using a dummy user (so that no messages get excluded)?
Also do you think we should add something like a dropdown menu to choose the lang?

File: website/src/pages/admin/status/index.tsx
Comment (ml729) [reply website/src/pages/admin/status/index.tsx]: I removed it.

File: website/src/pages/api/admin/status.ts
Comment (ml729) [reply website/src/pages/api/admin/status.ts]: Done (I'm assuming you meant getBackendUserCore?)

File: website/src/pages/api/admin/status.ts
Comment (notmd) [inline website/src/pages/api/admin/status.ts:16]: Sorry for the bad code suggestion here. Currently using Promise.all will cause all 3 endpoints either success or error. We should handle the error of them separately. [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#using_promise.allsettled) should be used here.

Comment (ml729) [reply website/src/pages/api/admin/status.ts]: Gotcha, I made the change.

File: website/src/pages/api/admin/tasks_availability.ts
Comment (andreaskoepf) [reply website/src/pages/api/admin/tasks_availability.ts]: I still think using a hypothetical dummy user to make the availablity query is fine.

Comment (fozziethebeat) [reply website/src/pages/api/admin/tasks_availability.ts]: Is the prod instance guaranteed to have the dummy user?  If so that'll be safe.

At a later date we should add a backend endpoint that returns the general availability independent of user and language.  But either using the admin or the dummy user is fine

**Review by fozziethebeat — APPROVED** [review]
Works well enough as a start~ 🚢

**Comment by fozziethebeat** [comment]
Oh! before I can merge, please fix the build and tests

File: website/src/pages/api/admin/tasks_availability.ts
Comment (ml729) [reply website/src/pages/api/admin/tasks_availability.ts]: Ok, I changed it to use a dummy user. Currently it seems that when calling `/api/v1/tasks_availability` the `PromptRepository` will create the user if they don't exist.

I agree about adding the user independent backend endpoint later.

**Comment by ml729** [comment]
I removed the imports that broke the build.

**Review by fozziethebeat — APPROVED** [review]

**Pull request merged**
